content/powershell/concepts/functions/functions.md

```diff
@@ -1,11 +1,121 @@
 ---
 Title: 'Functions'
-Description: ''
+Description: 'A function is a set of instructions that are executed together whenever invoked via its name. Parameters and importing modules can further enhance the usage of functions.'
 Subjects:
+  - 'Code Foundations'
   - 'Computer Science'
+  - 'Bash/Shell'
 Tags:
+  - 'Arguments'
   - 'Functions'
+  - 'Parameters'
 CatalogContent:
   - 'learn-powershell'
   - 'paths/computer-science'
 ---
+
+A **function** is a set of statements that are executed when the function name is called. Functions allow programmers to break down a problem into smaller chunks and help define a particular task. A few of the major reasons for using functions are readability and reusability.
+
+## Function Definition
+
+The syntax for defining functions in PowerShell is:
+
+```pseudo
+Function <Name>{
+  <statement 1>
+  <statement 2>
+  <statement 3>
+  ...
+}
+```
+
+The `Name` of the function is used to invoke the function. Statements are written between the curly brackets `{ }`.
+
+## Invoking a Function
+
+To invoke or call a function, simply use the function's name.
+
+```shell
+Function StatusDisplay{
+  $date = get-date -format "MM/dd/yyyy HH:mm:ss"
+  write-host "Script is running at $date"
+}
+
+StatusDisplay
+# Output: Script is running at 02/27/2023 06:24:04
+```
+
+## Parameters and Arguments
+
+Parameters can be added to function declaration to make them more flexible. The keyword `Param` is used to declare all parameters in the first function statement.
```

Comment:
```suggestion
Parameters can be added to the function declaration to make them more flexible. The keyword `param` is used to declare all parameters in the first function statement.
```

Reply:
Fixed in 23f54309efceaf758722ab60f2a5dde33e0f9e65